### PR TITLE
Wip/remove pangolin dependency

### PR DESCRIPTION
**Pull request by gocarlos**
Changes made: 
- removed most files which are not needed for mono SLAM
- some changes to the cmakelists (e.g. new dependences for opencv, gcc etc. )
- removed pangolin need when compiling.

### CODE REVIEW

**Comment by nidegen** [comment]
hmm does not compile on mac yet. ok if i adapt it for mac first of mergint? also, i could add macos ci before merging

**Comment by gocarlos** [comment]
Jep, could you do that? I will eventually get a mac later this month....

**Comment by nidegen** [comment]
sure, aber morn morge erst:)

**Review by nidegen — APPROVED** [review]
lgtm

i merged it with my osx version and everything builds fine. thanks!

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+dist: trusty
```
Comment (nidegen) [inline .travis.yml:1]: @+1 for travis:)

**Pull request merged**